Phase 3, Add templates and JS example for Enriching Errors content (#1855)
getsentry/sentry-docs: src/_includes/common/errors-context.md

{% comment %}
Guideline: This page is comprehensive; it is stored in the common folder, nested under _includes/common. To use, 

If you have questions, please ask Fiona or Daniel. 

> 1. If you haven't already, add the errors content folder to directory of the platform you are documenting -- _documentation/sdks/<sdk/platform>/errors (for example, _documentation/sdks/javascript/errors). 

Author: Thanks.

3. Create the defined `include` statements and add them to the errors-context.md file

If you have questions, please ask Fiona or Daniel. 

**The objective for this page is that a developer can easily view the what context is with events for this SDK**
{% endcomment %}

Data is sent with every event, and is either predefined or custom:

- **Predefined data** is data that Sentry recognizes. This data enhances your ability to understand and investigate the source and impact of the data when viewing it in the Sentry web UI. For example, in Sentry, the concept of user helps Sentry display the number of unique users affected by an event.
- **Custom data** is arbitrary structured or unstructured extra data you can attach to your event.

Regardless of whether the data sent to Sentry is predefined or custom, additional data can take two forms, tags and context:

- *Tags* are key/value string pairs that are both indexed and searchable. Tags power UI features such as filters and tag-distribution maps. Tags help you quickly access related events and view the tag distribution for a set of events. Common uses for tags include hostname, platform version, and user language.

>     Once you've started sending tagged data, you'll see it in the Sentry web UI: the filters within the sidebar on the Project page; summarized within an event; and on the tags page for an aggregated event.

Author: Thanks.

### Predefined Data

Sentry turns additional, predefined data or specific attributes on the data into _tags_, which you can use for a variety of purposes, such as searching in the web UI for the event or delving deeper into your application. For example, you can use tags such as `level` or `user.email` to surface particular errors. You can also enable Sentry to track [releases](/workflow/releases/), which unlocks features that help you delve more deeply into the differences between deployed releases.

If Sentry captures some predefined data but doesn’t expose it as a tag, you can set a custom tag for it.

Request, device, OS, runtime, app, browser, GPU, logger, and monitor are the most typical predefined data sent with an event. In addition, the following are sent, and can be modified for your team's use:

Level

: Defines the severity of an event. The level can be set to one of five values, which are, in order of severity: `fatal`, `error`, `warning`, `info`, and `debug.error`. Learn how to set the level in Set the Level

User

> : Providing user information to Sentry helps you evaluate the number of users affecting an issue and evaluate the quality of the application. Learn how to capture user information in Capture the User

Author: It will. Final fixes (x-links) are planned for the final phase.

Fingerprint

: Sentry uses one or more fingerprints to determine how to group errors into issues. Learn more about Sentry's approach to grouping algorithms and fingerprinting in Grouping & Fingerprints. Learn how to override the the default group in very advanced use cases in Modifying the Default Fingerprint

Environment

> : Environments help you better filter issues, releases, and user feedback in the Issue Details page of the web UI. Learn how to set and manage environments in Manage Environments

Reviewer: Is "Manage Environments" supposed to be linked to a page or section?

Author: yes. fixed. Thanks.

Release

> : A release is a version of your code that you deploy to an environment. When enabled, releases also help you determine regressions between releases and their potential source as discussed our content discussing [releases](/workflow/releases/).

Reviewer: ```suggestion
: A release is a version of your code that you deploy to an environment. When enabled, releases also help you determine regressions between releases and their potential source, as discussed in the [releases documentation](/workflow/releases/).
```

{{ include.errors_release_name }}
{% comment %}
Guideline: Create the `include` statement that provides SDK specific information
{% endcomment %}

## Modifying Defaults

Modify the defaults with any of the following:

### Set the Level

{{ include.errors_set_level }}
{% comment %}
Guideline: Create the `include` statement that provides SDK specific information
{% endcomment %}

### Capture the User

Users consist of a few critical pieces of information that construct a unique identity in Sentry. Each of these is optional, but one **must** be present for the Sentry SDK to capture the user:

id

: Your internal identifier for the user.

username

: The username. Typically used as a better label than the internal id.

email

: An alternative (or addition) to the username. Sentry is aware of email addresses and can display things such as Gravatars and unlock messaging capabilities.

ip_address

> : The user's IP address. If the user is unauthenticated, Sentry uses the IP address as a unique identifier for the user. Sentry will attempt to pull this from the HTTP request data, if available.

Reviewer: @MimiDumpling leaving note for myself so I know where I left off.

Author: Thanks for the edits thus far.
